Sources/performance-worker/WorkerService.swift

```diff
@@ -98,3 +164,45 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
     throw RPCError(code: .unimplemented, message: "This RPC has not been implemented yet.")
   }
 }
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension WorkerService {
+  private func setupServer(_ config: Grpc_Testing_ServerConfig) async throws -> GRPCServer {
+    let server = GRPCServer(transports: [], services: [BenchmarkService()])
+    let stats = try await ServerStats()
+
+    try self.state.withLockedValue { state in
+      try state.setupServer(server: server, stats: stats)
+    }
+
+    return server
+  }
+
+  private func makeServerStatsResponse(
+    reset: Bool
+  ) async throws -> Grpc_Testing_WorkerService.Method.RunServer.Output {
+    let server = self.state.withLockedValue { state in state.server }
+    guard server != nil else {
+      throw RPCError(code: .failedPrecondition, message: "This worker doesn't have a server setup.")
+    }
+    let currentStats = try await ServerStats()
+    let initialStats = self.state.withLockedValue { state in
+      return state.serverStats(replaceWith: currentStats)
```

Comment:
We're ignoring `reset` here: 

```suggestion
      return state.serverStats(replaceWith: reset ? currentStats : nil)
```